Adding additional descriptions

ramirezg opened the pull request:
8 of the 9 descriptions were added via the all_usa_contacts.json file. 

The descriptions for these agencies were not included in the first iteration because originally they were not top-level agencies due to a [manual data mapping issue](https://github.com/18F/foia/pull/74).

Looking back at this code, I realize that it might be possible to also add descriptions to many, many more non-top level agencies with some mapping work.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -42,7 +38,7 @@ def load_all_usa_data():
             data[office['Name']] = {
                 'description': office.get('Description', 'No Description'),
-                'id': office.get('Id', 'No Id'),
+                'id': office.get('Id'),
```
I don't understand the code changes here, and how they solve the problem. Come see me at  some point today?

ramirezg replied:
I changed these lines because it's painful to see my how ugly my first PR was. The real changes are to foiaHub-usaContacts-matches.xlsx Also, I'll drop by when I get in.

khandelwal replied:
These code changes are not related to the PR and just make the code cleaner.

khandelwal commented:
Let's see if these changes are covered by testing.

ramirezg commented:
The some of the changes were covered by testing, and this branch should have actually failed. (See nosetests below) The question is why didn't it?! 

Also, many of our functions have poor test coverage, maybe we should start fixing this...

```bash
FAILED (failures=1)
(foia)GSA-xs-MacBook-Pro:contacts gabrielramirez$ nosetests --with-coverage
.............F.............................................
======================================================================
FAIL: extract one acroym, if it has two return 'massive error'
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gabrielramirez/Documents/FOIA/foia/contacts/tests/layer_with_usa_contacts_tests.py", line 21, in test_extract_acronym
    "", usa_layer.extract_acronym("Random Office"))
AssertionError: '' != None

Name                        Stmts   Miss  Cover   Missing
---------------------------------------------------------
_markupbase                   276    276     0%   8-395
cgi                           581    581     0%   12-1061
csv                           260    260     0%   2-449
ctypes                        333    333     0%   3-541
ctypes._endian                 33     33     0%   1-59
ctypes.macholib                 1      1     0%   9
ctypes.macholib.dyld           88     88     0%   5-158
ctypes.macholib.dylib          20     20     0%   5-63
ctypes.macholib.framework      23     23     0%   5-65
ctypes.util                   180    180     0%   1-267
encodings.idna                180    180     0%   3-301
html                           37     37     0%   5-132
html.entities                   8      8     0%   4-2506
html.parser                   337    337     0%   11-550
http.client                   679    679     0%   69-1302
http.cookiejar               1097   1097     0%   28-2077
http.cookies                  175    175     0%   39-578
json                           43     43     0%   98-331
json.decoder                  209    209     0%   3-362
json.encoder                  228    228     0%   3-433
json.scanner                   56     56     0%   3-73
keywords_from_fr              126     70    44%   53-65, 101-107, 113-116, 131-154, 160-166, 172-204, 208-209
layer_with_csv                115     44    62%   79-82, 87, 109-124, 149-177, 180-181
layer_with_reading_room       119     42    65%   14-17, 102-115, 147-160, 166-169, 173-181
layer_with_usa_contacts        78     47    40%   33-43, 48-62, 67-84, 110-122, 127-128
mimetypes                     193    193     0%   26-592
netrc                         106    106     0%   5-142
processing_time_scraper       181     81    55%   29, 49, 93-113, 120-128, 137-140, 146-160, 186-196, 203, 242-243, 256-260, 266-285, 290-291
scraper                       288     43    85%   164, 232-233, 248, 268, 323-325, 390-394, 423-444, 457, 462-463, 474-475, 485-494
shlex                         233    233     0%   9-304
sqlite3                         1      1     0%   23
sqlite3.dbapi2                 43     43     0%   23-89
stringprep                     65     65     0%   2-272
typos                           1      0   100%   
unittest.mock                1245   1245     0%   7-2330
uuid                          307    307     0%   47-591
---------------------------------------------------------
TOTAL                        7945   7364     7%   
----------------------------------------------------------------------
Ran 59 tests in 1.135s

FAILED (failures=1)

```

ramirezg commented:
@khandelwal , I'm thinking of holding off on this PR till we start using the live API and a cleaner mapper.

ramirezg commented:
Solves 18F/foia-hub/issues/470
Additional updates:
1. Cleaned up unused files.
2. Improved testing
3. Moved 8 manual descriptions out of the excel spreadsheet into the manual files.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -1,132 +1,168 @@
+    (' - FOIA Program Office (ARSF)', ''),
+    (' - Main Office', ''),
+    (' Activity', ''),
+    (' - Headquarters Office', ''),
```
Some comments around why these are in here, or where we should expect to see them would be helpful.

ramirezg replied:
Okay, I added in more descriptive comments.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -1,132 +1,168 @@
+def extract_abbreviation(name):
+    """ Extract abbreviation from name string """
+
+    match = ACRONYM.search(name)
```
Python functions return None by default. If you want less code you can do: 

```python
if match:
  return match.group(0).strip("() ")
```

and not have to do the abbreviation=None bit.

ramirezg replied:
Okay, I cleaned up this code.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -1,132 +1,168 @@
+    new_dict = {'usa_id': data['Id']}
+    abbreviation = extract_abbreviation(name=data['Name'])
+    if abbreviation:
+        new_dict.update({'abbreviation': abbreviation})
```
You don't have to call update() you can just do

```python
new_dict['abbreviation'] = abbrevation
```

ramirezg replied:
no idea why I used update.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -1,132 +1,168 @@
+
+def create_contact_dict(data):
+    """
+    Generates a dictionary containing a usa id, description, and
```
What's a 'usa id'?

ramirezg replied:
It's the id in the USA contacts API, we were keeping it before.

khandelwal replied:
Ah - but my point was - we should probably document that reasonably well.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -1,132 +1,168 @@
+
+    new_dict = {}
+    for contact_data in data:
+        if contact_data['Language'] == "en":
```
Are there other Languages? Why are we making this check here?

ramirezg replied:
Yeah, many of the records are translated into Spanish, so this limits them to English.

khandelwal replied:
Add that in as a comment.

khandelwal commented on contacts/layer_with_usa_contacts.py:
```diff
@@ -1,132 +1,168 @@
-        with open(filename, 'w') as f:
-            f.write(yaml.dump(
-                yaml_data, default_flow_style=False, allow_unicode=True))
+def main():
```
Instead of calling this main, call it something more meaningful.

khandelwal commented on contacts/tests/layer_with_usa_contacts_tests.py:
```diff
@@ -1,48 +1,174 @@
+            usa_layer.clean_name(name_1), usa_layer.clean_name(name_2))
+
+    def test_extract_abbreviation(self):
+        """ Test if abbrivations are extracted properly from name """
```
abbreviations

ramirezg replied:
Fixed

khandelwal commented:
This PR now uses the usa.gov API with a cache. Should you update the README.md or some documentation somewhere to indicate what one needs to do if they want to clear the cache and hit the API again?

ramirezg commented:
@khandelwal I'll update the documentation and make the suggested code changes.

ramirezg commented:
Made the corrections and updated the [README](https://github.com/18F/foia/pull/103/files#diff-5ff56ff296baa2ee2c3fa6505133337bL3). It still needs a lot of work so I started an issues, which is now in the backlog.

khandelwal commented on contacts/README.md:
```diff
@@ -1,18 +1,26 @@
 [![Coverage Status](https://coveralls.io/repos/18F/foia/badge.png)](https://coveralls.io/r/18F/foia)
 
-## Agency FOIA contact info
+## FOIA Contact Data Builder
```
Contacts

khandelwal commented on contacts/README.md:
```diff
@@ -1,18 +1,26 @@
+## FOIA Contact Data Builder
 
-Downloading agency contact information for FOIA, from FOIA.gov.
+The scripts in this directory are used for constructing a FOIA contact database for Federal Agencies and Offices. The data collected are stored
```
s/database/dataset

khandelwal commented on contacts/README.md:
```diff
@@ -1,18 +1,26 @@
-Downloading agency contact information for FOIA, from FOIA.gov.
+The scripts in this directory are used for constructing a FOIA contact database for Federal Agencies and Offices. The data collected are stored
+in a set of yaml files inside the `data/` directory. These yaml files are
+then uploaded to 18F's [Django application](https://github.com/18F/foia-hub)
```
They are not uploaded in this repository - so let's find a clearer way to say this.

ramirezg replied:
I'm just going to delete "via [this script](https://github.com/18F/foia-hub/blob/master/foia_hub/scripts/load_agency_contacts.py)." I think it suffices to say that the data are for the Django app.

khandelwal commented on contacts/README.md:
```diff
@@ -31,15 +39,27 @@ python keywords_from_fr.py
+layer_with_usa_contacts.py -> usa_contacts.sqlite
+processing_time_scraper.py -> html/
+keywords_from_fr.py -> fr.sqlite
+layer_with_reading_room.py -> None
```
Remove this line. It's confusing.
